### ISSUE TITLE
Create a tvOS/iOS/etc app?

### ISSUE THREAD

**Issue by chriseidhof** [chriseidhof/pomotv#31]
It'd be really cool to have an app for this (as several people have noted on Twitter). I think this could be a great app to build with Functional View Controllers. Anyone who feels like taking this on, let me know, maybe I can help!

**Comment by tomlokhorst** [comment]
Yeah, it would be cool to have a tvOS app. Now I'm using AirPlay from my Mac, but an app is nicer.

Unfortunately, I don't think there are supported ways for playing YouTube or Vimeo videos on tvOS.
Although there is https://github.com/hellozimi/HCYoutubeParser, that is not allowed under the YouTube Terms of Service, so it might stop working.

**Comment by chriseidhof** [comment]
Hm, and there's no web view, right? Is there a YouTube app on tvOS that we could link to?

**Comment by BalestraPatrick** [comment]
We could use https://github.com/0xced/XCDYouTubeKit to play videos in tvOS. The project says to support tvOS but I don't have any experience with it.

**Comment by tomlokhorst** [comment]
Today, I hacked together an iOS app: https://github.com/Q42/pomotv-app

Because there's no API (https://github.com/chriseidhof/pomotv/issues/11) I've used the `recents.xml` Atom feed. The app doesn't show all videos since there are only a few youtube links in the recents file. However I think its better to use an actual API instead of putting work into parsing the feeds.

The code isn't very nice, but it's playing videos. Also I got distracted, because I was watching one of the talks.

**Comment by chriseidhof** [comment]
Very cool! I think @johankj was thinking of adding an API. What would you need? I think we could probably just present the JSON variant of the current data in the .yml files, right?

**Comment by tomlokhorst** [comment]
Yes, JSON version of the current data would be nice.

I just now realised the search is happening in JavaScript, since this is a static site.
I'll have a look at the index (http://www.pomo.tv/search/index.json) to see if I can use that as well.

**Comment by DerLobi** [comment]
I hacked up a [tvOS app](https://github.com/DerLobi/pomotv-tvos-client). It currently displays and plays all youtube and vimeo videos, and shows the videos of the most recent event on the top shelf.

Here are some screenshots:
![tvOS app](https://github.com/DerLobi/pomotv-tvos-client/raw/master/screenshots.png)

**Comment by chriseidhof** [comment]
Oh, awesome! What are your plans with this? Maybe we could put this on the website or in the README once you're comfortable with it?

**Comment by DerLobi** [comment]
That would be cool. My only concern is that I currently use `XCDYoutubeKit` and I don't really know what this would mean in the worst case (would the app be rejected? could YouTube sue me?). I have tried to use `UIApplication.sharedApplication().openURL(youtubeURL)` with the YouTube app installed but it doesn't work on tvOS.

Besides that I would like to reach feature parity with the website (browse events, speakers, tags), get WWDC videos to play and clean up the code.

I'd like to offer the app as a free download from the App Store if (legally) possible. If that doesn't work maybe keep it available on GitHub only. Because the target audience is iOS developers we could assume they are able to build and run the app themselves.

**Comment by eelco** [comment]
@DerLobi What would you like in terms of an API? (#11) I see you’re now using the YAML from the repository, ideally ‘something’ should be exposed directly at pomo.tv itself to make it a bit less fragile.

From https://github.com/0xced/XCDYouTubeKit:

> XCDYouTubeKit is against the YouTube [Terms of Service](https://www.youtube.com/t/terms)

If I understand it correctly, the app would violate 4.A and users of the app would violate 4.C. The consequence could be termination of your account (7.A). But, more importantly, this also means the app would/should not pass app review, since the [guidelines](https://developer.apple.com/app-store/review/guidelines/) clearly state that “Apps may not [...] violate 3rd party terms of use” (8.5)

It might be ‘best’ to create a radar and ask for `openURL` support and submit a request to YouTube for an SDK :disappointed: For now, keeping it available on GitHub is of course great :+1:

**Comment by DerLobi** [comment]
if you could expose the data from the YAML files as JSON that would be great.
Also for the recently added videos I currently get the last event from `editions.yml` and show all corresponding videos. Would be cool to have the feed XML as a JSON array too. Then could show the same videos on the top shelf as are currently shown on the home page.

`openURL` generally works on tvOS (so no radar necessary), but the YouTube app does not support any of the URL schemes I tried.

**Comment by SusannProszak** [comment]
Any news about the api? @DerLobi do you need help with the tvOS app? Saw that there are still some topics left to implement.

**Comment by chriseidhof** [comment]
There is no news on this, it should be quite easy to generate JSON straight from the YAML. Is anyone interested in doing this?
